projects/swimlane/ngx-charts/src/lib/box-chart/box.component.ts

```diff
@@ -0,0 +1,381 @@
+        .attr('y1', (d, index, node) => this.lineCoordinates[index].v1.y)
+        .attr('x2', (d, index, node) => this.lineCoordinates[index].v2.x)
+        .attr('y2', (d, index, node) => this.lineCoordinates[index].v2.y);
+
+      // Looking for a short way of perform each attr transition.
+      // lineEl.each((d, i, node) => this.lineTween('x1', d, i, node));
+    } else {
+      lineEl
+        .attr('x1', (d, index, node) => this.lineCoordinates[index].v1.x)
+        .attr('y1', (d, index, node) => this.lineCoordinates[index].v1.y)
+        .attr('x2', (d, index, node) => this.lineCoordinates[index].v2.x)
+        .attr('y2', (d, index, node) => this.lineCoordinates[index].v2.y);
+    }
+    this.oldLineCoordinates = this.lineCoordinates;
+  }
+
+  /**
+   * See [D3 Selections](https://www.d3indepth.com/selections/)
+   * @param d The joined data.
+   * @param index The index of the element within the selection
+   * @param node The node element (Line).
+   */
+  lineTween(attr: string, d: any, index: number, node: BaseType[] | ArrayLike<BaseType>) {
+    const nodeLineEl = node[index] as SVGLineElement;
+    console.log('NodeEL: ', nodeLineEl);
+    console.log(`- ${attr} `, nodeLineEl[attr].baseVal.value);
+    // console.log('- X1: ', nodeLineEl.x1.baseVal.value);
+    // console.log('- Y1: ', nodeLineEl.y1.baseVal.value);
+    // console.log('- X2: ', nodeLineEl.x2.baseVal.value);
+    // console.log('- Y2: ', nodeLineEl.y2.baseVal.value);
+    return nodeLineEl[attr].baseVal.value;
+  }
+
+  // TODO: Refactor into another .ts file if https://github.com/swimlane/ngx-charts/pull/1179 gets merged.
+  pathTween(d1: string, precision: number) {
+    return function () {
+      // tslint:disable-next-line: no-this-assignment
+      const path0 = this;
+      const path1 = this.cloneNode();
+      const n0 = path0.getTotalLength();
+      // tslint:disable-next-line: ban-comma-operator
+      const n1 = (path1.setAttribute('d', d1), path1).getTotalLength();
+
+      // Uniform sampling of distance based on specified precision.
+      const distances = [0];
+      let i = 0;
+      const dt = precision / Math.max(n0, n1);
+      while (i < 1) {
+        distances.push(i);
+        i += dt;
+      }
+      distances.push(1);
+
+      // Compute point-interpolators at each distance.
+      const points = distances.map((t: number) => {
+        const p0 = path0.getPointAtLength(t * n0);
+        const p1 = path1.getPointAtLength(t * n1);
+        return interpolate([p0.x, p0.y], [p1.x, p1.y]);
+      });
+
+      // 't': T is the fraction of time (between 0 and 1) since the transition began.
+      return (t: any) => {
+        return t < 1 ? 'M' + points.map((p: (t: number) => any[]) => p(t)).join('L') : d1;
+      };
+    };
+  }
+
+  getStartingPath(): string {
+    if (!this.animations) {
+      return this.getPath();
+    }
+
+    let path = '';
+
+    if (this.roundEdges) {
+      if (this.orientation === 'vertical') {
```

Comment:
pls use https://github.com/swimlane/ngx-charts/blob/6d591944a7eaa2942b4b4c39c9d3526d28ceb1b9/projects/swimlane/ngx-charts/src/lib/common/types/orientation.enum.ts#L1

Reply:
same applies below

Reply:
Are you sure this is Orientation? As the options from that enum ain't "Vertical" neither "Horizontal" as the ones here. I think it should be the same as `BarOrientation` from the `bar.component.ts`.